inspire_json_merger/pre_filters.py

```diff
@@ -113,6 +116,35 @@ def filter_publisher_references(root, head, update):
     return root, head, update
 
 
+def update_authors_with_ordering_info(root, head, update):
+    """Adds ordering information into authors entries.
+
+    This is helpful so we won't loose correct ordering on conflicts.
+    Args:
+        root (pmap): the root record.
+        head (pmap): the head record.
+        update (pmap): the update record.
+
+    Returns: A tuple containing root, head and update with authors data
+        enriched with ordering information
+
+    """
+    if root.get("authors"):
+        root = root.update({"authors": _update_authors_list_with_ordering_data(root["authors"])})
+    if head.get('authors'):
+        head = head.update({"authors": _update_authors_list_with_ordering_data(head["authors"])})
+    if update.get('authors'):
+        update = update.update({"authors": _update_authors_list_with_ordering_data(update["authors"])})
+    return root, head, update
```

Comment:
I don't think this is correct, as you may have collisions between the ordering info from the different parts. Do you really need ordering info everywhere? Isn't it sufficient to have it only in the `head`?

Reply:
If it will be only in head then we will have many differencec in merger (it will try to update/remove this ORDER_KEY)

Reply:
I don't think so, as it doesn't conflict with anything.

Reply:
Still without ordering info everywhere it will be really difficult to merge this few HEAD entries with merged result. With ordering info everywhere is really simple and fast.